### PR TITLE
Change some configurations, include change the defaultChannelNum to 16

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo [xinguo.li@zilliz.com](mailto:xinguo.li@zilliz.com)
#23475

### CODE REVIEW

File: configs/milvus.yaml [hunk 1]
```diff
@@ -115,14 +115,14 @@ rocksmq:
   # please adjust in embedded Milvus: /tmp/milvus/rdb_data
   path: /var/lib/milvus/rdb_data
   lrucacheratio: 0.06 # rocksdb cache memory ratio
-  rocksmqPageSize: 268435456 # 256 MB, 256 * 1024 * 1024 bytes, The size of each page of messages in rocksmq
-  retentionTimeInMinutes: 7200 # 5 days, 5 * 24 * 60 minutes, The retention time of the message in rocksmq.
+  rocksmqPageSize: 67108864 # 64 MB, 64 * 1024 * 1024 bytes, The size of each page of messages in rocksmq
```
Comment (jiaoew1991) [inline configs/milvus.yaml:118]: Need to update the corresponding 'Doc' field in service_params.go.

File: internal/rootcoord/timeticksync.go [hunk 2]
```diff
@@ -111,8 +111,15 @@ func (c *chanTsMsg) getTimetick(channelName string) typeutil.Timestamp {
 }
 
 func newTimeTickSync(ctx context.Context, sourceID int64, factory msgstream.Factory, chanMap map[typeutil.UniqueID][]string) *timetickSync {
+	// if the old channels number used by the user is greater than the set default value currently
+	// keep the old channels
+	defaultChanNum := Params.RootCoordCfg.DmlChannelNum.GetAsInt64()
+	if int64(len(chanMap)) > defaultChanNum {
+		defaultChanNum = int64(len(chanMap))
+	}
```
Comment (congqixia) [inline internal/rootcoord/timeticksync.go:119]: `len(chanMap)` cannot be used as channel number. Length is actually collection number.

Comment (congqixia) [reply internal/rootcoord/timeticksync.go:119]: Also, can we use the exceeded channel numbers here?

**Review by congqixia — CHANGES_REQUESTED** [review]
Max Channel Number can be scaled in. What's the process of scaling in pre-configured channel names?

File: internal/rootcoord/util.go [hunk 3]
```diff
@@ -135,3 +135,15 @@ func getTravelTs(req TimeTravelRequest) Timestamp {
...
+func getNeedChanNum(setNum int, chanMap map[typeutil.UniqueID][]string) int {
+	var ret int
+
+	for _, chanNames := range chanMap {
+		ret += len(chanNames)
+	}
+	if setNum > ret {
+		ret = setNum
+	}
+	return ret
```
Comment (congqixia) [inline internal/rootcoord/util.go:148]: case 
chanMap = map[int64]channels {
1: {"dml_1", "dml_2"},
2: {"dml_2", "dml_3"},
}
needed number shall be 3, not 4

**Review by congqixia — CHANGES_REQUESTED** [review]

File: pkg/util/paramtable/service_param.go [hunk 4]
```diff
@@ -584,17 +584,17 @@ please adjust in embedded Milvus: /tmp/milvus/rdb_data`,
 	r.PageSize = ParamItem{
 		Key:          "rocksmq.rocksmqPageSize",
 		DefaultValue: strconv.FormatInt(64<<20, 10),
-		Version:      "2.0.0",
-		Doc:          "256 MB, 256 * 1024 * 1024 bytes, The size of each page of messages in rocksmq",
+		Version:      "2.2.7",
+		Doc:          "64 MB, 64 * 1024 * 1024 bytes, The size of each page of messages in rocksmq",
```
Comment (congqixia) [inline pkg/util/paramtable/service_param.go:588]: We shall not change the version for changing the default value

File: pkg/util/paramtable/service_param.go [hunk 5]
```diff
@@ -584,17 +584,17 @@ please adjust in embedded Milvus: /tmp/milvus/rdb_data`,
...
 	r.PageSize.Init(base.mgr)
 
 	r.RetentionTimeInMinutes = ParamItem{
 		Key:          "rocksmq.retentionTimeInMinutes",
-		DefaultValue: "7200",
-		Version:      "2.0.0",
-		Doc:          "5 days, 5 * 24 * 60 minutes, The retention time of the message in rocksmq.",
+		DefaultValue: "4320",
+		Version:      "2.2.7",
+		Doc:          "3 days, 3 * 24 * 60 minutes, The retention time of the message in rocksmq.",
```
Comment (congqixia) [inline pkg/util/paramtable/service_param.go:597]: ditto

**Review by congqixia — COMMENTED** [review]
overall lgtm
could you please fix the version update issue?

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/rootcoord/util.go [hunk 6]
```diff
@@ -135,3 +135,16 @@ func getTravelTs(req TimeTravelRequest) Timestamp {
...
 	return ts == typeutil.MaxTimestamp
 }
+
+func getNeedChanNum(setNum int, chanMap map[typeutil.UniqueID][]string) int {
+	chanNames := typeutil.NewSet[string]()
+	for _, chanName := range chanMap {
+		chanNames.Insert(chanName...)
+	}
+
+	ret := chanNames.Len()
```
Comment (xiaofan-luan) [inline internal/rootcoord/util.go:145]: what if chanMap is 
prefix-1, prefix-8, prefix-10?
I think we need to create at least 10 channels on that case

File: internal/rootcoord/dml_channels.go [hunk 7]
```diff
@@ -155,7 +155,7 @@ func newDmlChannels(ctx context.Context, factory msgstream.Factory, chanNamePref
 	// if topic created, use the existed topic
 	if params.PreCreatedTopicEnabled.GetAsBool() {
 		chanNamePrefix = ""
-		chanNum = int64(len(params.TopicNames.GetAsStrings()))
+		chanNum = int64(len(params.TopicNames.GetAsStrings())) + chanNumDefault
```
Comment (xiaofan-luan) [inline internal/rootcoord/dml_channels.go:158]: don't really understand it's meaning

**Pull request merged**